app/src/main/java/org/mozilla/focus/web/IWebView.java

```diff
@@ -75,6 +76,8 @@ public HitTarget(final boolean isLink, final String linkURL, final boolean isIma
         void resetBlockedTrackers();
 
         void onBlockingStateChanged(boolean isBlockingEnabled);
+
+        void onHttpAuthRequest(HttpAuthHandler handler, String host, String realm);
```

Comment:
HttpAuthHandler is a specific implementation for WebView. We'd need to hide that behind an interface that IWebView defines.